tests/cpp/test_multidevice_communications.cpp

```diff
@@ -407,6 +410,53 @@ TEST_P(CommunicationTest, ReduceScatter) {
   }
 }
 
+TEST_P(CommunicationTest, AllgatherLoopSplit) {
```

Comment:
tests/cpp/test_multidevice_lower_communication.cpp seems to be a better place for this test.

Reply:
I am using the current file since the test setup runs for both NCCL and UCC which is important when modifying the postCommunication calls and ensure the changes are compatible with both interfaces.

Reply:
> UCC

I understand your intention. Can you double check UCC indeed kicked in for this particular test? For other tests in this file, `backend_` is created according to the test parameter and is then directly used by postSingleCommunication. However, I doubt `backend_` affects FusionExecutorCache at all. 

Instead, I'd extend test_multidevice_lower_communication.cpp to test UCC as well as NCCL.